index.js

```diff
@@ -73,32 +73,30 @@ class USHINBase {
       );
     }
 
-    const allPoints = new Set();
-
-    if (main) allPoints.add(main);
-
-    for (const shape of Object.keys(shapes)) {
-      const pointIds = shapes[shape];
-      for (const pointId of pointIds) {
-        const point = pointStore[pointId];
-        if (!point) {
-          const error = new Error("Point ID not found in store");
-          error.pointId = pointId;
-          throw error;
-        }
-        if (!point._id) throw new Error("Must specify point ID");
-        if (!point._rev) {
-          await this.addPoint({ createdAt: createdAtTime, ...point });
-        }
-        allPoints.add(pointId);
-        if (point.referenceHistory) {
-          for (const { pointId: referencePoint } of point.referenceHistory) {
-            allPoints.add(referencePoint);
-          }
-        }
+    const allPoints = new Set([main, ...Object.values(shapes).flat()]);
+
+    for (const pointId of allPoints) {
+      const point = pointStore[pointId];
+      if (!point) {
+        const error = new Error("Point ID not found in store");
+        error.pointId = pointId;
+        throw error;
+      }
+      if (!point._id) throw new Error("Must specify point ID");
+      if (!point._rev) {
+        await this.addPoint({ createdAt: createdAtTime, ...point });
       }
     }
 
+    // Convert set to array to avoid iterating over pointIds which are
+    // added inside the conditional block below
+    [...allPoints].forEach((pointId) => {
+      const point = pointStore[pointId];
+      if (point.referenceHistory) {
+        point.referenceHistory.forEach((log) => allPoints.add(log.pointId));
```

Comment:
Is there an advantage of using `.forEach` instead of `for of`? It might be good to be consistent.

Reply:
I'm just more accustomed to it so that's what I used, but consistency is probably more important. It's not a big deal for me at all :)